File: bob/Bob.cs
using System;
using System.Linq;

public static class Bob
{
    public static bool IsQuestion(string statement) { return statement.EndsWith("?"); }
    
    public static bool IsYelling(string statement) { return statement.ToUpper() == statement && statement.Any(char.IsLetter); }
    
    public static bool IsStatementEmpty(string statement) { return String.IsNullOrWhiteSpace(statement); }
    
    public static string Response(string statement)
    {
        string trimmedStatement = statement.Trim();
        if (IsQuestion(trimmedStatement))
        {
            if (IsYelling(trimmedStatement)) return "Calm down, I know what I'm doing!";
            else return "Sure.";
        } 
        else if (IsYelling(trimmedStatement)) return "Whoa, chill out!";
        else if (IsStatementEmpty(trimmedStatement)) return "Fine. Be that way!";
        else return "Whatever.";
    }
}

File: bob/bob.ts
function isQuestion(message: string): boolean {
  return message.endsWith("?");
}

function isYelling(message: string): boolean {
  return message.split("").every((e) => e === e.toUpperCase());
}

function doesContainLetters(message: string): boolean {
  return message
    .split("")
    .some((e) => "abcdefghijklmnopqrstuvwxyz".includes(e.toLowerCase()));
}

export function hey(message: string): string {
  message = message.trim();
  if (doesContainLetters(message)) {
    if (isYelling(message) && isQuestion(message)) {
      return "Calm down, I know what I'm doing!";
    } else if (isYelling(message) && !isQuestion(message)) {
      return "Whoa, chill out!";
    } else if (isQuestion(message)) {
      return "Sure.";
    } else if (isYelling(message)) {
      return "Whoa, chill out!";
    }
  } else if (message === "") {
    return "Fine. Be that way!";
  } else if (isQuestion(message)) {
    return "Sure.";
  }
  return "Whatever.";
}
